cli/popper/utils.py

```diff
@@ -2,147 +2,67 @@
         return t
 
     ecode = None
-    # quick shortcut for 1) above
-    if not verbose and not log_file:
-        out = ""
-        if debug:
-            info('DEBUG: Using subprocess.check_output() for {}\n'.format(cmd))
-        try:
-            out = check_output(cmd, shell=True, stderr=PIPE,
-                               universal_newlines=True)
-            ecode = 0
-        except CalledProcessError as ex:
-            ecode = ex.returncode
-            if debug:
-                info('DEBUG: Catched exception: {}\n'.format(ex))
-            if not ignore_error:
-                fail("Command '{}' failed: {}\n".format(cmd, ex))
-        return b(out).strip(), ecode
 
     sleep_time = 0.25
     num_times_point_at_current_sleep_time = 0
-    outf = None
-    errf = None
-
-    if log_file:
-        if verbose:
-            if debug:
-                info('\nDEBUG: Creating file for combined stdout/stderr\n')
-            outf = open(log_file + '.log', 'w')
-        else:
-            if debug:
-                info('\nDEBUG: Creating separate files for stdout/stderr\n')
-            outf = open(log_file + '.out', 'w')
-            errf = open(log_file + '.err', 'w')
 
     try:
-        if verbose:
-            if debug:
-                info('DEBUG: subprocess.Popen() with combined stdout/stderr\n')
-            p = Popen(cmd, stdout=PIPE, stderr=STDOUT, shell=True,
-                      universal_newlines=True, preexec_fn=os.setsid)
-        else:
-            if debug:
-                info('DEBUG: subprocess.Popen() with separate stdout/stderr\n')
-            p = Popen(cmd, stdout=outf, stderr=errf, shell=True,
-                      universal_newlines=True, preexec_fn=os.setsid)
+        log.debug('subprocess.Popen() with combined stdout/stderr')
+        p = Popen(cmd, stdout=PIPE, stderr=STDOUT, shell=True,
+                  universal_newlines=True, preexec_fn=os.setsid)
 
         if add_to_process_list:
             popper.cli.process_list.append(p.pid)
 
-        if debug:
-            info('DEBUG: Reading process output\n')
-
-        while ecode is None:
+        log.debug('Reading process output')
 
-            if verbose:
-                # read until end of file (when process stops)
-                for line in iter(p.stdout.readline, ''):
-                    line_decoded = b(line)
-                    info(line_decoded)
-                    if log_file:
-                        outf.write(line)
-            else:
-                # when we are not writing output to stdout, print dot progress
-                if sleep_time < 30 \
-                        and num_times_point_at_current_sleep_time == 5:
-                    sleep_time *= 2
-                    num_times_point_at_current_sleep_time = 0
+        for line in iter(p.stdout.readline, ''):
+            line_decoded = b(line)
+            log.info(line_decoded[:-1])
+        # when we are not writing output to stdout, print dot progress
```

Comment:
since the above loop consumes all the output, the block below (the one that used to sleep before; lines 43-53) are not needed anymore. As a consequence, variables `sleep_time` and `num_times_point_at_current_sleep_time` are also not needed.

Also, since this function is only used by the `HostRunner`, we can move it to `gha.py` as

Reply:
@ivotron Updated it